### PR TITLE
[WIP] Modify OSM.org sync to pull OSM XML and import it into a fresh DB

### PR DESCRIPTION
**Pull request by noffle**
This is the collector app half of an optimization to get sync working a bit
faster until osm-p2p-db is a bit quicker at syncing larger datasets.

Also addresses https://github.com/osmlab/field-data-collection/issues/141

Coordinator app PR: https://github.com/osmlab/field-data-coordinator/pull/30

### CODE REVIEW

File: app/lib/asyncstorage-chunk-store.js [hunk 1]
```diff
@@ -72,8 +72,21 @@ Storage.prototype.get = function(index, opts, cb) {
 
 Storage.prototype.close = Storage.prototype.destroy = function(cb) {
```
Comment (sethvincent) [inline app/lib/asyncstorage-chunk-store.js]: should we also make `close` a separate method that just sets `this.closed = true`?

Comment (noffle) [inline app/lib/asyncstorage-chunk-store.js]: Yes, let's!

**Comment by noffle** [comment]
I haven't tested on a device to compare, but they should be the same order of magnitude as [these results](https://github.com/osmlab/field-data-collection/issues/66#issuecomment-312373790) (importer vs p2p-sync).

**Comment by noffle** [comment]
I *think* this is good to go, but I haven't been able to fix whatever broke with my React Native install. I've tested locally by injecting a filesystem-backed osm-p2p-db, but I don't know whether the AsyncStorage db backend works.

@sethvincent Do you have the bandwidth to test this out against the coordinator app? The test procedure would be

1. Load in a new OSM.org area into the coordinator app
2. Load up a (fresh) collector app
3. Sync collector to coordinator; record a dump of AsyncStorage (somehow?)
4. Sync to coordinator again (don't clear anything); record another dump of AsyncStorage

**Comment by noffle** [comment]
Made one more change: we need to close and reopen to the DB *again* in order to trigger the osm-p2p-db indexes to start regenerating. Oh the technical debt I sow! :boom: :crying_cat_face: :memo:

**Comment by noffle** [comment]
Woohoo! Thank you for rocking this @sethvincent :heart: :tada: :taco:

**Pull request merged**